consensus/src/dag/reconstruction/mod.rs

```diff
@@ -72,8 +82,31 @@ impl<U: Unit> WrappedUnit<U::Hasher> for ReconstructedUnit<U> {
 }
 
 impl<U: Unit> UnitWithParents for ReconstructedUnit<U> {
-    fn parents(&self) -> &NodeMap<HashFor<Self>> {
-        &self.parents
+    fn parents(&self) -> impl Iterator<Item = &HashFor<U>> {
+        self.parents.values().map(|(hash, _)| hash)
+    }
+
+    fn direct_parents(&self) -> impl Iterator<Item = &HashFor<Self>> {
+        self.parents
+            .values()
+            .filter_map(|(hash, parent_round)| match self.unit.coord().round() {
+                0 => None,
+                unit_round => {
+                    if unit_round - 1 == *parent_round {
+                        Some(hash)
+                    } else {
+                        None
+                    }
+                }
+            })
+    }
+
+    fn parent_for(&self, index: NodeIndex) -> Option<&HashFor<Self>> {
+        self.parents.get(index).map(|(hash, _)| hash)
+    }
+
+    fn parents_size(&self) -> NodeCount {
```

Comment:
Better call this `node_count` or something, the current name might suggest that it's about the number of parents, which would be confusing.